lib/chef/knife/mixin/helper.rb

```diff
@@ -59,7 +59,7 @@ def validate_json(json)
             next unless printable?(value.to_s)
```

Comment:
shouldn't printable just be changed to also match whitespace?

Reply:
Hi @lamont-granquist , for the whitespace it is working fine without any issue as shwon below
```
json file content:
{
	"id": "test1",
	"text1": "my passwords",
	"username": "root"
}

bundle exec knife vault update test test1 -J ../../chef-repo/root.json -c ../../chef-repo/.chef/knife.rb

bundle exec knife vault show test test1 -c ../../chef-repo/.chef/knife.rb
id:       test1
text1:    my passwords
username: root


```
Based on this is there a need to add warnings for whitespace? It should be be fine here to not add warning IMO. Please suggest.

Reply:
I'm talking about adding `[[:space:]]` to the match for `printable?` which also matches `\n` and `\r` and `\t` in addition to ascii space and various unicode spaces.

Reply:
you probably want to just use `String#valid_encoding?` though.  or use the `String#encode!` method to convert the string to UTF-8 but raise on any errors rather than replacing.

Reply:
> space

okk, will add `[[:space]]`. Thanks @lamont-granquist !

Reply:
> I'm talking about adding `[[:space:]]` to the match for `printable?` which also matches `\n` and `\r` and `\t` in addition to ascii space and various unicode spaces.

Okk, will add `[[:space:]]` Thanks @lamont-granquist !